include/seqan3/range/views/minimiser_hash.hpp

```diff
@@ -0,0 +1,148 @@
+    }
+    //!\brief Store the shape, the window_size and the seed and return a range adaptor closure object.
+    constexpr auto operator()(seqan3::shape const & shape, uint32_t const & window_size, uint64_t const & seed) const
+    {
+        return seqan3::detail::adaptor_from_functor{*this, shape, window_size, seed};
+    }
+
+    /*!\brief            Call the view's constructor with the underlying view, a seqan3::shape and a window size as
+     *                   argument.
+     * \param[in] urange The input range to process. Must model std::ranges::viewable_range and the reference type
+     *                   of the range must model seqan3::semialphabet.
+     * \param[in] shape_ The seqan3::shape to use for hashing.
+     * \throws std::invalid_argument if the size of the shape is greater than the window_size.
+     * \returns          A range of converted elements.
+     */
+    template <std::ranges::range urng_t>
+    constexpr auto operator()(urng_t && urange, seqan3::shape const & shape, uint32_t const & window_size,
+                              uint64_t const & seed = 0x8F3F73B5CF1C9ADE) const
+    {
+        static_assert(std::ranges::viewable_range<urng_t>,
+            "The range parameter to views::minimiser_hash cannot be a temporary of a non-view range.");
+        static_assert(std::ranges::forward_range<urng_t>,
+            "The range parameter to views::minimiser_hash must model std::ranges::forward_range.");
+        static_assert(semialphabet<reference_t<urng_t>>,
+            "The range parameter to views::minimiser_hash must be over elements of seqan3::semialphabet.");
+
+        if (shape.size() > window_size)
+            throw std::invalid_argument{"The size of the shape cannot be greater than the window size."};
+
+        return std::forward<urng_t>(urange) | seqan3::views::kmer_hash(shape)
+                                            | std::views::transform([seed] (int i) { return i ^ seed; })
+                                            | seqan3::views::minimiser(window_size - shape.size() + 1);
+    }
+};
+//![adaptor_def]
+
+} // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\name Alphabet related views
+ * \{
+ */
+
+/*!\brief                    Computes minimisers for a range with a given shape, window size and seed.
+ * \tparam urng_t            The type of the range being processed. See below for requirements. [template parameter is
+ *                           omitted in pipe notation]
+ * \param[in] urange         The range being processed. [parameter is omitted in pipe notation]
+ * \param[in] shape          The seqan3::shape that determines how to compute the hash value.
+ * \param[in] window_size    The window size to use.
+ * \param[in] seed           The seed to use. Default: 0x8F3F73B5CF1C9ADE.
+ * \returns                  A range of std::integral where each value is the minimiser of the resp. window.
```

Comment:
```suggestion
 * \returns                  A range of `size_t` where each value is the minimiser of the resp. window.
```

Since we actually know the type?